examples/mplot3d/wire3d_animation_demo.py

```diff
@@ -1,38 +1,45 @@
-from __future__ import print_function
 """
-A very simple 'animation' of a 3D plot
+A very simple 'animation' of a 3D plot.  See also rotate_axes3d_demo.
 """
+
+from __future__ import print_function
+
 from mpl_toolkits.mplot3d import axes3d
 import matplotlib.pyplot as plt
 import numpy as np
 import time
 
 
 def generate(X, Y, phi):
+    '''
+    Generates Z data for the points in the X, Y meshgrid and parameter phi.
+    '''
     R = 1 - np.sqrt(X**2 + Y**2)
     return np.cos(2 * np.pi * X + phi) * R
 
+
 fig = plt.figure()
 ax = fig.add_subplot(111, projection='3d')
 
+# Make the X, Y meshgrid.
 xs = np.linspace(-1, 1, 50)
 ys = np.linspace(-1, 1, 50)
 X, Y = np.meshgrid(xs, ys)
-Z = generate(X, Y, 0.0)
 
+# Set the z axis limits so they aren't recalculated each frame.
+ax.set_zlim(-1, 1)
```

Comment:
I'm just realizing I was thinking of the rotate_axes3d example when I wrote the note saying the z limits changed 'as the rotation progressed'.  Sorry for that confusion.  In fact it's the data that's changing in this example, not the rotation.  As the largest z value contracts from say 1 to 0.6, the z limits are self-adjusting so that the extremes of the surface are always using the full z range.  In that updated context, would you say that the behaviour is as expected or does something still sound fishy?